packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -265,6 +246,12 @@ export class OdspDocumentService implements IDocumentService {
                     this.socketIoClientFactory(),
                 ]);
 
+            // Start refreshing join session periodically. 
+            if (!this.isJoinSessionPeriodicallyRefreshing) {
```

Comment:
I find usage around thisJoinSessionPeriodicallyRefreshing not ideal. Basically, what you do is ensure there is only one instance of refreshSessionPeriodically() running per document session. That's good.
But the way code is written, it has couple not ideal side-effects:
1. Loop in refreshSessionPeriodically() will keep running even when we do not have connection (i.e. disconnected and can't connect back).
2. On reconnection, it will not take into account new refreshSessionDurationSeconds value, it will continue its cycle, and thus likely will call joinSessison sooner than needed.

I think the better flow is the one where
- on lost connection, we stop refreshSessionPeriodically() loop and get out
- on new connection we start it again.